base/boot.jl

```diff
@@ -253,6 +254,9 @@ end
 struct ArgumentError <: Exception
     msg::AbstractString
 end
+struct UnassignedKeyword <: Exception
```

Comment:
Should this with `Error` or `Exception`?

Reply:
Sure, I can do that.   `UnassignedKeywordError` is a bit long.  Maybe `UndefKeywordError`?

Reply:
`KeywordError`? Or `KeywordException`?

Reply:
We also have `UndefVarError` and `UndefRefError`, so `UndefKeywordError` seems consistent.  `KeywordError` seems too vague, since there are other ways you could get errors from keyword args.

Reply:
But `UndefKeywordError` feels a bit like something that would be thrown for cases like
```julia
foo(;a = 1) = a
foo(b=2)
```

Reply:
I would consider that an unexpected keyword argument error – "undef" is pretty consistently used to mean something that is used as a value without having been given one.